File: 11 opencv（二）/threshold.cpp
//import cv2
//
//img = cv2.imread("1.jpg")
//gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
//ret, binary = cv2.threshold(gray, 0, 255, cv2.THRESH_BINARY | cv2.THRESH_OTSU)
//
//cv2.imshow("gray",gray)
//cv2.imshow('binary', binary)
//cv2.waitKey(0)

#include<opencv2/opencv.hpp>

int main() {
    cv::Mat img = cv::imread("1.jpg");

    cv::Mat gray_img;
    cv::cvtColor(img,gray_img,cv::COLOR_BGR2GRAY);

    cv::Mat bin_img;
    cv::threshold(gray_img,bin_img,0.,255.,cv::THRESH_BINARY | cv::THRESH_OTSU);

    cv::imshow("gray_img", gray_img);
    cv::imshow("bin_img",bin_img);
    cv::waitKey(0);
}

File: 11 opencv（二）/test20210119/threshold.py
import cv2
import numpy as np
from matplotlib import pyplot as plt
img = cv2.imread('1.jpg',0)
ret,thresh1 = cv2.threshold(img,0,255,cv2.THRESH_BINARY+cv2.THRESH_OTSU)
ret,thresh2 = cv2.threshold(img,127,255,cv2.THRESH_BINARY)
# ret,thresh3 = cv2.threshold(img,150,255,cv2.THRESH_BINARY)
cv2.imshow("Binary1", thresh1)
cv2.imshow("Binary2", thresh2)
# cv2.imshow("Binary3", thresh3)

cv2.waitKey()
# ret,thresh2 = cv2.threshold(img,127,255,cv2.THRESH_BINARY_INV)
# ret,thresh3 = cv2.threshold(img,127,255,cv2.THRESH_TRUNC)
# ret,thresh4 = cv2.threshold(img,127,255,cv2.THRESH_TOZERO)
# ret,thresh5 = cv2.threshold(img,127,255,cv2.THRESH_TOZERO_INV)
# titles = ['Original Image','BINARY','BINARY_INV','TRUNC','TOZERO','TOZERO_INV']
# images = [img, thresh1, thresh2, thresh3, thresh4, thresh5]
# for i in range(6):
#     plt.subplot(2, 3, i + 1), plt.imshow(images[i], 'gray')
#     plt.title(titles[i])
#
# plt.show()